src/utils/transactions.rs

```diff
@@ -52,15 +61,29 @@ pub(super) fn hash(safe_address: Address, nonce: u64) -> [u8; 32] {
     keccak256(encoded)
 }
 
-pub(super) fn domain_hash(safe_address: &Address) -> [u8; 32] {
+pub(super) fn domain_hash(safe_address: &Address, is_legacy: bool) -> [u8; 32] {
+    let domain_separator_typehash = if is_legacy {
+        DOMAIN_SEPARATOR_TYPEHASH_LEGACY
+    } else {
+        DOMAIN_SEPARATOR_TYPEHASH
+    };
```

Comment:
Wondering if the type hash would change in the future we would end up with more legacy references. What if we match the version here (not sure if it's possible). Was thinking something along the lines:

```rust
let safe_version = Version::parse(safe_info.version.unwrap().as_str());
let domain_separator_typehash = match safe_version {
  Ok(Version::parse(SAFE_V_1_3_0)) => { DOMAIN_SEPARATOR_TYPEHASH }
  Ok(_) => { DOMAIN_SEPARATOR_TYPEHASH_LEGACY }
  Err(_) => {} // panic/error result in case we cannot retrieve the version?
};
```

Reply:
I would not do this as in your code any newer version would also use the legacy typehash. But for version 1.4.0 we most likely will have the same typehash as for 1.3.0

Reply:
I'd actually remove the default that I mentioned and be more explicit on the versions in the matching or throw an error otherwise

Reply:
Also doesn't need to be for each specific version but if the branches can be interval of versions I think that would also work :)

Reply:
I think that makes sense, but I would match the interval and not the `Result` , I think that if we get an `Err` we would fallback to the legacy hash, right?

Reply:
Not sure what the best behaviour here would be for the fallback. I'm more leaning toward an error an being more conservative regarding the version matching. But this is mostly a suggestion! It looks good :)